http4s/src/main/scala/com/avast/grpc/jsonbrige/http4s/Http4s.scala

```diff
@@ -4,67 +4,60 @@ import cats.data.NonEmptyList
 
-  def apply[F[_]: Sync](configuration: Configuration)(bridges: GrpcJsonBridge[F, _ <: BindableService]*): HttpRoutes[F] = {
+  def apply[F[_]: Sync](configuration: Configuration)(bridge: GrpcJsonBridge[F]): HttpRoutes[F] = {
     implicit val h: Http4sDsl[F] = Http4sDsl[F]
     import h._
 
-    val bridgesMap = bridges.map(s => (s.serviceName, s): (String, GrpcJsonBridge[F, _])).toMap
-
     val pathPrefix = configuration.pathPrefix
       .map(_.foldLeft[Path](Root)(_ / _))
       .getOrElse(Root)
 
-    logger.info(s"Creating HTTP4S service proxying gRPC services: ${bridges.map(_.serviceName).mkString("[", ", ", "]")}")
+    logger.info(s"Creating HTTP4S service proxying gRPC services: ${bridge.servicesNames.mkString("[", ", ", "]")}")
 
     val http4sService = HttpRoutes.of[F] {
       case _ @GET -> `pathPrefix` / serviceName if serviceName.nonEmpty =>
-        bridgesMap.get(serviceName) match {
-          case Some(service) =>
-            Ok {
-              service.methodsNames.mkString("\n")
-            }
+        NonEmptyList.fromList(bridge.methodsNames.filter(_.service == serviceName).toList) match {
           case None =>
             val message = s"Attempt to GET non-existing service: $serviceName"
             logger.debug(message)
             NotFound(message)
+          case Some(methods) =>
+            Ok { methods.map(_.fullName).toList.mkString("\n") }
         }
 
       case _ @GET -> `pathPrefix` =>
-        Ok { bridgesMap.values.flatMap(s => s.methodsNames).mkString("\n") }
+        Ok { bridge.methodsNames.map(_.fullName).mkString("\n") }
 
       case request @ POST -> `pathPrefix` / serviceName / methodName =>
         val headers = request.headers
         headers.get(`Content-Type`.name) match {
           case Some(Header(_, contentTypeValue)) =>
             `Content-Type`.parse(contentTypeValue) match {
               case Right(`Content-Type`(MediaType.application.json, _)) =>
-                bridgesMap.get(serviceName) match {
-                  case Some(service) =>
-                    request
-                      .as[String]
-                      .flatMap(service.invokeGrpcMethod(methodName, _, mapHeaders(headers)))
-                      .flatMap {
-                        case Right(resp) => Ok(resp, `Content-Type`(MediaType.application.json))
-                        case Left(st) => mapStatus(st, configuration)
-                      }
-                  case None =>
-                    val message = s"Attempt to POST non-existing service: $serviceName"
-                    logger.debug(message)
-                    NotFound(message)
-                }
+                request
+                  .as[String]
+                  .flatMap(
+                    body =>
+                      bridge.invoke(GrpcMethodName(serviceName, methodName),
+                                    body,
+                                    request.headers.toList.map(h => (h.name.value, h.value)).toMap))
```

Comment:
Why you don't extract `request.headers.toList.map(h => (h.name.value, h.value)).toMap)` to own method (like it was until now)?